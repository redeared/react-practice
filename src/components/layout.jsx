import React from 'react';

const flexColumnContainer = {
    display: 'flex',
    flexDirection: 'column',
    flex: '1 1 auto',
};
export const ColumnsLayout = ({ style, children, ...rest }) => (
    <div style={{ ...flexColumnContainer, ...style }} {...rest} >
        {children}
    </div>
);

const flexRowContainer = {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    height: '100%',
};
export const RowsLayout = ({ style, children, ...rest }) => (
    <div style={{ ...flexRowContainer, ...style }} {...rest} >
        {children}
    </div>
);

export const GrowLayoutBlock = ({ style, children, ...rest }) => (
    <div style={{ flexGrow: '1', ...style }} {...rest} >
        {children}
    </div>
);

export const FixedLayoutBlock = ({ style, basis, children, ...rest }) => (
    <div style={{ flexBasis: basis || 0, ...style }} {...rest} >
        {children}
    </div>
);
