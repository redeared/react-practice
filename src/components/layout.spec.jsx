import React from 'react';
import TestRenderer from 'react-test-renderer';

import { ColumnsLayout, RowsLayout, FixedLayoutBlock, GrowLayoutBlock } from './layout';


describe('Layout components', () => {
    it('GrowLayoutBlock should render without crash', () => {
        const comp = TestRenderer.create(<GrowLayoutBlock>child</GrowLayoutBlock>).toJSON();
        expect(comp).toMatchSnapshot();
    });

    it('RowsLayout should render without crash', () => {
        const comp = TestRenderer.create(<RowsLayout>child</RowsLayout>).toJSON();
        expect(comp).toMatchSnapshot();
    });
});
