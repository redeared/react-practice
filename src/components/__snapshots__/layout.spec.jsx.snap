// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout components GrowLayoutBlock should render without crash 1`] = `
<div
  style={
    Object {
      "flexGrow": "1",
    }
  }
>
  child
</div>
`;

exports[`Layout components RowsLayout should render without crash 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": "100%",
    }
  }
>
  child
</div>
`;
